interpreter/convert.go

```diff
@@ -18,39 +24,44 @@ func convertToStarlark(value interface{}) (starlark.Value, error) {
 	case uint64:
 		return starlark.MakeUint64(v), nil
 	case float32:
+		return starlark.Float(v), nil
 	case float64:
 		return starlark.Float(v), nil
 	case string:
 		return starlark.String(v), nil
-	case []interface{}:
-		var elems []starlark.Value
-		for _, elem := range v {
-			val, err := convertToStarlark(elem)
-			if err != nil {
-				return nil, err
-			}
-			elems = append(elems, val)
-		}
-		return starlark.NewList(elems), nil
-	case map[interface{}]interface{}:
-		dict := starlark.NewDict(len(v))
-
-		for index, elem := range v {
-			key, err := convertToStarlark(index)
-			if err != nil {
-				return nil, err
+	default:
+		reflectV := reflect.ValueOf(value)
+		switch reflectV.Kind() {
+		case reflect.Slice:
+			var elems []starlark.Value
+			for i := 0; i < reflectV.Len(); i++ {
+				val, err := convertToStarlark(reflectV.Index(i).Interface())
+				if err != nil {
+					return nil, errors.Wrapf(err, "failed to convert slice element %d", i)
+				}
+				elems = append(elems, val)
 			}
+			return starlark.NewList(elems), nil
+		case reflect.Map:
+			dict := starlark.NewDict(len(reflectV.MapKeys()))
 
-			val, err := convertToStarlark(elem)
-			if err != nil {
-				return nil, err
-			}
+			iter := reflectV.MapRange()
+			for iter.Next() {
+				key, err := convertToStarlark(iter.Key().Interface())
+				if err != nil {
+					return nil, errors.Wrapf(err, "failed to convert map key %s", key.String())
+				}
 
-			if err = dict.SetKey(key, val); err != nil {
-				return nil, err
+				val, err := convertToStarlark(iter.Value().Interface())
+				if err != nil {
+					return nil, errors.Wrapf(err, "failed to convert map val %s: %s", key.String(), val.String())
+				}
+				if err = dict.SetKey(key, val); err != nil {
+					return nil, err
+				}
 			}
+			return dict, nil
 		}
-		return dict, nil
 	}
-	return nil, nil
+	return nil, fmt.Errorf("cannot convert value %#v", value)
```

Comment:
for now we should change the first return vals (on errors) to be `starlark.None` if nil gets returned it gets v v unhappy.
```suggestion
	return starlark.None, fmt.Errorf("cannot convert value %#v", value)
```